packages/custom_lint/test/mock_fs.dart

```diff
@@ -9,15 +9,25 @@ import 'dart:io';
 Future<T> runWithIOOverride<T>(
   FutureOr<T> Function(Stream<String> out, Stream<String> err) testBody, {
   Directory? currentDirectory,
+  bool supportsAnsiEscapes = false,
 }) async {
   final fs = _MockFs(
     stdout,
     currentDirectory ?? Directory.current,
+    supportsAnsiEscapes: supportsAnsiEscapes,
   );
 
   try {
     return await IOOverrides.runWithIOOverrides(
-      () => testBody(fs.stdout.stream, fs.stderr.stream),
+      () => runZoned(
+        () => testBody(fs.stdout.stream, fs.stderr.stream),
+        zoneSpecification: ZoneSpecification(
+          print: (self, parent, zone, line) {
```

Comment:
This sounds fishy. Why do we need this?

Reply:
It'd make debugging pretty tedious too

Reply:
Well there is a custom `StreamController` which `stdout/stderr` is piped to, to create expectations.
Prints are not piped into the custom controller otherwise.

Reply:
But that's on purpose

Reply:
This is removed now